tests/test_avro_schema.py

```diff
@@ -8,6 +8,206 @@
 import json
 import pytest
 
+# Schemas defined in AvroCompatibilityTest.java. Used here to ensure compatibility with the schema-registry
+schema1 = parse('{"type":"record","name":"myrecord","fields":[{"type":"string","name":"f1"}]}')
+schema2 = parse(
+    '{"type":"record","name":"myrecord","fields":[{"type":"string","name":"f1"},{"type":"string",'
+    '"name":"f2","default":"foo"}]}'
+)
+schema3 = parse('{"type":"record","name":"myrecord","fields":[{"type":"string","name":"f1"},{"type":"string","name":"f2"}]}')
+schema4 = parse('{"type":"record","name":"myrecord","fields":[{"type":"string","name":"f1_new","aliases":["f1"]}]}')
+schema6 = parse('{"type":"record","name":"myrecord","fields":[{"type":["null","string"],"name":"f1","doc":"doc of f1"}]}')
+schema7 = parse(
+    '{"type":"record","name":"myrecord","fields":[{"type":["null","string","int"],"name":"f1","doc":"doc of f1"}]}'
+)
+# schema8 = (
+#     '{"type":"record","name":"myrecord","fields":[{"type":"string","name":"f1"},{"type":"string",'
+#     '"name":"f2","default":"foo"}]},{"type":"string","name":"f3","default":"bar"}]}'
+# )
+badDefaultNullString = parse(
+    '{"type":"record","name":"myrecord","fields":[{"type":["null","string"],"name":"f1","default":'
+    '"null"},{"type":"string","name":"f2","default":"foo"},{"type":"string","name":"f3","default":"bar"}]}'
+)
+
+
+def test_schemaregistry_basic_backwards_compatibility():
+    """
+    Backward compatibility: A new schema is backward compatible if it can be used to read the data
+    written in the previous schema.
+    """
+    msg = "adding a field with default is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema2, schema1)
+    assert res.compatibility is SchemaCompatibilityType.compatible, msg
+
+    msg = "adding a field w/o default is NOT a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema3, schema1)
+    assert res.compatibility is SchemaCompatibilityType.incompatible, msg
+
+    # msg = "changing field name with alias is a backward compatible change"
+    # res = ReaderWriterCompatibilityChecker().get_compatibility(schema4, schema1)
+    # assert res.compatibility is SchemaCompatibilityType.compatible, msg
```

Comment:
Would be nice if you included the `locations` and `messages` in the failure message, so we could pinpoint the problem

Reply:
I added an automatic formatter for this. It will looks like this:

```
    assert res != SchemaCompatibilityResult.compatible(), msg                                                                                                                                                                                 
E   AssertionError: adding a field with default is a backward compatible change                                                                                                                                                               
E   assert Comparing SchemaCompatibilityResult instances:                                                                                                                                                                                     
E       Left:                                                                                                                                                                                                                                 
E         compatibility SchemaCompatibilityType.compatible                                                                                                                                                                                    
E         locations:                                                                                                                                                                                                                          
E           /                                                                                                                                                                                                                                 
E       Right:                                                                                                                                                                                                                                
E         compatibility SchemaCompatibilityType.compatible                                                                                                                                                                                    
E         locations:... 
```